k8s/eks/bash/functions.sh

```diff
@@ -0,0 +1,346 @@
+  echo "ebs=$ebs_volume" >> $real_path/.cluster/$CLUSTER_NAME.cs
+}
+
+make_persistent_volume(){  
+  export TG_EBS_DATADIR_VOLUME_ID=$ebs_volume
+
+  EBS_PV=$(mktemp)
+  envsubst <$real_path/yaml/ebs/pv.yml.spec >$EBS_PV
+  kubectl apply -f $real_path/yaml/ebs/storageclass.yml -f $EBS_PV -f $real_path/yaml/ebs/pvc.yml
+}
+
+helm_redis_add_repo(){
+  helm repo add bitnami https://charts.bitnami.com/bitnami
+} 
+
+helm_infra_install_redis(){
+  helm install testground-infra-redis --set auth.enabled=false --set master.nodeSelector='testground.node.role.infra: "true"' bitnami/redis
+} 
+
+tg_daemon_config_map(){
+  kubectl create -f - <<EOF
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: env-toml-cfg
+  namespace: default
+data:
+  .env.toml: |
+    ["aws"]
+    region = "$REGION"   
+
+    [runners."cluster:k8s"]
+    run_timeout_min             = 15
+    testplan_pod_cpu            = "100m"
+    testplan_pod_memory         = "100Mi"
+    collect_outputs_pod_cpu     = "1000m"
+    collect_outputs_pod_memory  = "1000Mi"
+    provider                    = "aws"
+    sysctls = [
+      "net.core.somaxconn=10000",
+    ]
+
+    [daemon]
+    listen = "0.0.0.0:8042"
+    slack_webhook_url = ""
+    github_repo_status_token = ""
+
+    [daemon.scheduler]
+    workers = 2
+    task_timeout_min  = 20
+    task_repo_type    = "disk"
+
+    [client]
+    endpoint = "localhost:8080"
+
+EOF
+}
+
+tg_daemon_service_account(){
+  kubectl apply -f $real_path/yaml/tg-daemon-service-account.yml
+}
+
+tg_daemon_role_binding(){
+  kubectl apply -f $real_path/yaml/tg-daemon-cluster-role.yml
+}
+
+tg_daemon_services(){
+  export CLUSTER_NAME="$CLUSTER_NAME"
+
+  TG_DAEMON_SVC=$(mktemp)
+  envsubst <$real_path/yaml/tg-daemon-svc.yml.spec >$TG_DAEMON_SVC
+  kubectl apply -f $TG_DAEMON_SVC
+}
+
+tg_daemon_svc_sync_service(){
+  kubectl apply -f $real_path/yaml/tg-sync-service.yml
+}
+
+tg_daemon_sync_service(){
+  kubectl apply -f $real_path/yaml/tg-sync-service-deployment.yml
+}
+
+tg_daemon_sidecar(){
+  kubectl apply -f $real_path/yaml/tg-ds-sidecar.yml
+}
+
+tg_daemon_deployment(){
+  kubectl apply -f $real_path/yaml/tg-daemon-deployment.yml
+}
+
+obtain_alb_address(){
+  ALB_ADDRESS=$(kubectl get services -l app=testground-daemon -o jsonpath="{.items[0].status.loadBalancer.ingress[0].hostname}")
+}
+
+obtain_alb_name(){
+  ALB_NAME=$(kubectl get services -l app=testground-daemon -o jsonpath="{.items[0].status.loadBalancer.ingress[0].hostname}" | cut -d'-' -f1)
+}
+
+wait_for_alb_and_instances(){
+  obtain_alb_name
+  aws elb wait any-instance-in-service --load-balancer-name $ALB_NAME --region $REGION
+}
+
+echo_env_toml(){
+  echo "Your 'testground/.env.toml' file needs to look like this:"
+  echo ""
+  echo "["aws"]"
+  echo "region = \"$REGION\""
+  echo "[client]"
+  echo "endpoint = \"http://$ALB_ADDRESS:80\""
+  echo 'user = "YOUR_NAME"'
```

Comment:
```suggestion
  echo 'user = "$(whoami)"'
```

Reply:
Modified the function so it outputs `whoami` but in a slightly different way.